tools/crm_node.c

```diff
@@ -194,6 +235,57 @@ node_name_xml(pcmk__output_t *out, va_list args) {
     return pcmk_rc_ok;
 }
 
+PCMK__OUTPUT_ARGS("partition-list", "GList *")
+static int
+partition_list_default(pcmk__output_t *out, va_list args)
+{
+    GList *nodes = va_arg(args, GList *);
+
+    GString *buffer = NULL;
+
+    for (GList *node_iter = nodes; node_iter != NULL; node_iter = node_iter->next) {
+        pcmk_controld_api_node_t *node = node_iter->data;
+        if (pcmk__str_eq(pcmk__s(node->state, ""), "member", pcmk__str_none)) {
+            pcmk__add_separated_word(&buffer, 128, pcmk__s(node->uname, ""), " ");
```

Comment:
I'm thinking there's no point in adding a space if we don't have uname, but I'm not sure what we should add. Maybe "unknown-node" or "node-id-%u"?

Reply:
What circumstances would lead us to not have a uname?

We could perhaps just skip adding anything including the space, by just checking that node->uname is not NULL first.

Reply:
There are a few rare but possible scenarios. We still support Corosync 2, which can have nodes that are learned by ID first and name later. I believe there are some corner cases with Corosync 3 as well but I don't remember them. We should always know at least one of name and ID.

It's an open question whether we should list anything. On the one hand, the request is to list cluster members, and it is a member (and having an unknown member could be a useful thing to know). On the other hand, if the user wants to parse node names, anything would be unhelpful. Safest bet is to preserve previous behavior, but I'm not sure it's desirable.

Reply:
In light of the attrd_updater thing, I'm a little hesitant to change behavior no matter how weird it seems.